internal/providers/ionoscloud/ionoscloud.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2019 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// The IONOSCloud provider fetches configurations from the userdata available in
+// the injected file /var/lib/cloud/seed/nocloud/user-data.
+// NOTE: This provider is still EXPERIMENTAL.
+
+package ionoscloud
+
+import (
+	"os"
+
+	"github.com/coreos/ignition/v2/config/v3_6_experimental/types"
+	"github.com/coreos/ignition/v2/internal/platform"
+	"github.com/coreos/ignition/v2/internal/providers/util"
+	"github.com/coreos/ignition/v2/internal/resource"
+
+	"github.com/coreos/vcontext/report"
+)
+
+const (
+	defaultFilename = "/var/lib/cloud/seed/nocloud/user-data"
```

Comment:
This won't work. Ignition runs from the initramfs and the `/var` there is different. You could inject it in what _would_ become `/var` in the real root, but you'd have to special-case code for CoreOS which feels really like a layering violation.

At least for FCOS, probably the simplest way to hack around this _for now_ (until there's a proper metadata service) is to inject the Ignition config in the boot partition (GPT partition label `boot`) at `/ignition/config.ign`. I'm not sure if there's an equivalent on the Flatcar side. We wouldn't even need to touch Ignition for that; the QEMU qcow2 image we publish should work for now.

Reply:
Yep, we just had a chat and realized this fact as well. In case of Flatcar the root partition is mounted to /sysroot/ and therefore the path is wrong anyway. I'll try to build a flatcar image using a patch version of ignition and we need to investigate how to make it work on other distributions and make the base path configurable

Reply:
@tormath1 am I right on this on? Regarding the `/sysroot` being a mount point of the boot partition?